src/platform/Linux/CHIPBluezHelper.cpp

```diff
@@ -864,86 +850,54 @@ gboolean BluezPeripheralRegisterApp(void * apClosure)
  * GATT Characteristic object
  ***********************************************************************/
 
-static void BluezHandleAdvertisementFromDevice(BluezDevice1 * aDevice)
+static void BluezHandleAdvertisementFromDevice(BluezDevice1 * aDevice, BluezEndpoint * endpoint)
 {
     const char * address   = bluez_device1_get_address(aDevice);
+    const char * flags     = bluez_device1_get_advertising_flags(aDevice);
     GVariant * serviceData = bluez_device1_get_service_data(aDevice);
-    GVariant * entry;
-    GVariantIter iter;
-    BluezAddress * src;
-    const uint8_t * tmpBuf;
-    uint8_t * buf;
-    size_t len, dataLen;
-    size_t i, j;
-    uint16_t serviceId;
-    char * debugStr;
 
-    // service data is optional and may not be present
-    SuccessOrExit(serviceData != nullptr);
-
-    src = g_new0(BluezAddress, 1);
+    GVariantIter serviceIterator;
+    GVariant * serviceEntry;
+    chip::Ble::ChipBleUUID uuid;
+    chip::Ble::ChipBLEDeviceIdentificationInfo deviceInfo;
+    char * debugStr = nullptr;
+    size_t dataLen;
 
-    BluezStringAddressToCHIPAddress(address, src);
+    // service data is optional and may not be present
+    VerifyOrExit(serviceData != nullptr, (void) 0);
```

Comment:
In OT we do have possibility to just call: `VerifyOrExit(serviceData != nullptr)`, what about just:
`VerifyOrExit(serviceData != nullptr, );`

Reply:
VerifyOrExit(serviceData != nullptr, ) works fine, but I thought it looked unfinished :). I'll change it and let the community decide.